price-feeder/oracle/oracle.go

```diff
@@ -2,29 +2,38 @@ package oracle
 
 import (
 	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"math"
 	"sync"
 	"time"
 
+	rpcClient "github.com/cosmos/cosmos-sdk/client/rpc"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
-
+	"github.com/umee-network/umee/price-feeder/oracle/client"
+	"github.com/umee-network/umee/price-feeder/oracle/provider"
 	pfsync "github.com/umee-network/umee/price-feeder/pkg/sync"
+	umeetypes "github.com/umee-network/umee/x/oracle/types"
+	"google.golang.org/grpc"
 )
 
 type Oracle struct {
-	logger zerolog.Logger
-	closer *pfsync.Closer
-
+	logger          zerolog.Logger
+	closer          *pfsync.Closer
 	mtx             sync.RWMutex
 	lastPriceSyncTS time.Time
 	prices          map[string]sdk.Dec
+	oracleClient    *client.OracleClient
 }
 
-func New() *Oracle {
+func New(oc *client.OracleClient) *Oracle {
```

Comment:
We need to provide the `CurrencyPairs` from the config as another argument here and create a "reverse flattened" map of provider => exchange pairs, i.e. a `map[string]ExchangePair`.

Reply:
I took a slightly different approach and used a `map[string][]string`, so I could just pass the string array into the `GetTickerPrices` method